src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -238,7 +244,9 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
       this._playerTracks = this._getParsedTracks();
       this._addNativeAudioTrackChangeListener();
       this._addNativeTextTrackChangeListener();
-      this._addNativeTextTrackAddedListener();
+      if (!(this._config && this._config.playback && this._config.playback.useNativeTextTrack)) {
```

Comment:
Move the if statement inside the `_addNativeTextTrackAddedListener `
BTW, if u want to check deep hierarchy of existing object plz use `Utils.Object` methods

Reply:
it shouldn't, we use it in a lot of different places. Check again what goes wrong or talk to me and show me what flow says.